src/scanner.ts

```diff
@@ -1,33 +1,100 @@
-
 //
 // The scanner's entire exposed functionality, independent
 // of the HTTP calls.
 //
 
-import * as sqlite3 from 'sqlite3'
-import {Election} from './types'
-import {addBallot} from './store'
-import * as interpreter from './interpreter'
+import * as chokidar from 'chokidar'
+import * as path from 'path'
+import * as fs from 'fs'
+import * as streams from 'memory-streams'
+import { Ballot, Election } from './types'
+import { addBallot, exportCVRs } from './store'
+import interpretFile from './interpreter'
 
-export function configure(election:Election) {
-  // start watching the ballots
-  interpreter.init(election, "./ballots", addBallot)
+import exec from './exec'
+
+export const ballotsPath = path.join(__dirname, '..', 'ballots/')
+export const scannedBallotsPath = path.join(__dirname, '..', 'scanned-ballots/')
+export const sampleBallotsPath = path.join(__dirname, '..', 'sample-ballots/')
+
+// make sure those directories exist
+const allPaths = [ballotsPath, scannedBallotsPath]
+allPaths.forEach(path => {
+  if (!fs.existsSync(path)) {
+    fs.mkdirSync(path)
+  }
+})
+
+// keeping track of election
+let watcher: chokidar.FSWatcher, election: Election
+
+// keeping track of status
+let status = '' // scanning, exporting, idle
+let message = ''
+let numBallotsScanned = 0
+
+function setStatus(newStatus: string, newMessage: string) {
+  status = newStatus
+  message = newMessage
+}
+
+export function fileAdded(ballotPath: string) {
+  interpretFile(election, ballotPath, (ballotPath: string, ballot: Ballot) => {
```

Comment:
I'm concerned about confusion occurring at some point due to the name of the variable changing between `votes` (see: https://github.com/votingworks/module-scan/pull/5/files#diff-a22679c0e1167fd703b79102f0b220adR94) and `ballot` for the second argument of the callback.

Reply:
good call, I will fix this.